src/components/auth/CreateAccount.js

```diff
@@ -1,118 +1,145 @@
-                                type="password"
-                                name="password"
-                                className="createAccountInput"
-                                id="password"
-                                placeholder="Password"
-                                required="" />
-                            <input onChange={this.handleFieldChange}
-                                type="password"
-                                name="confirmPassword"
-                                className="createAccountInput"
-                                id="confirmPassword"
-                                placeholder="Confirm password"
-                                required="" />
-
-                        </div>
-                        <button onClick={this.createNewUser} type="submit" className="button" id="signupButton">
-                            Register
-                        </button>
-                        <Link className="smallLink" to="/login">I already have an account</Link>
-                    <picture>
-                        <img src={Cassette} alt="cassette tape drawing" id="cassetteImage"/>
-                    </picture>
-                    </fieldset>
-                </form>
-            </>
-        )
-    }
+import React, { useReducer } from 'react'
+import ApiManager from '../modules/ApiManager'
+import { Link } from 'react-router-dom'
+import Logo from '../../images/logo.png'
+import Cassette from '../../images/cassette.png'
+import './Auth.css'
 
+const initialState = {
+  firstName: '',
+  lastName: '',
+  email: '',
+  password: '',
+  confirmPassword: '',
 }
 
-export default CreateAccount;
\ No newline at end of file
+const reducer = (state, action) => {
+  switch (action.type) {
+    case 'setField':
+      return {
+        ...state,
+        [action.field]: action.value,
+      }
+    default:
+      return state
+  }
+}
+
+/**
+ *
+ * @name AccountCreate - this should be named something more like this. CreateAccount implies that it is a utility rather than a Component. We can discuss this later
+ * @description Account creation form. A stateful component which holds the user's registration data, validates it, and processes it using the ApiManager
+ *
+ * @notes
+ *  - It has been refactored to be a functional component using the `useReducer` hook for state management.
+ *  - Functions and conditionals have been broken up into reasonably sized smaller parts.
+ *  - Variables (except the Component name) have been renamed to try to explain exactly what they do, since now they should do one or two things.
+ *  - Promises have been refactored to use modern ES7 "async/await" syntax to clean up the `.then` callback chaining.
+ *  - We are programatically giving the `onFieldChange` function its variables, rather than inferring them from element attributes. JavaScript FTW!
+ */
+export const CreateAccount = ({ setUser, history }) => {
```

Comment:
Here we are destructuring the props. Similar to the comment below about destructuring state, this makes our code more readable/understandable by explicitly stating what props we are going to be expecting for this component. The next step would be to implement `PropTypes` (I recommend starting with this), even further would be to use Flow or TypeScript (don't do that yet). 

It also allows us to save having to write `props.` like what you see below compared to the code in this file. 
```js
export const CreateAccount = (props) => {
....

    await props.setUser(results)
    await props.history.push('/')
```